vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewManager.java

```diff
@@ -0,0 +1,119 @@
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.services.ISpaceOverviewManager;
+
+@Service
+public class SpaceOverviewManager implements ISpaceOverviewManager {
+
+    @Autowired
+    private SpaceRepository spaceRepo;
+
+    @Autowired
+    private SpaceLinkDisplayRepository spaceLinkDisplayRepository;
+
+    @Autowired
+    private ModuleLinkDisplayRepository moduleLinkDisplayRepository;
+
+    /**
+     * This method is used to fetch all linked modules corresponding to a space.
+     */
+    @Override
+    public Map<String, List<ModuleLinkDisplay>> getSpaceToModuleLinks() {
+
+        Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap = new HashMap<>();
+        Iterable<ModuleLinkDisplay> spaceToModuleLinksList = moduleLinkDisplayRepository.findAll();
+
+        for (ModuleLinkDisplay link : spaceToModuleLinksList) {
+            String spaceId = null;
+
+            if (link.getLink() != null && link.getLink().getSpace() != null) {
+                spaceId = link.getLink().getSpace().getId();
+                if (!spaceToModuleLinksMap.containsKey(spaceId)) {
+                    spaceToModuleLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToModuleLinksMap.get(spaceId).add(link);
+            }
+        }
+        return spaceToModuleLinksMap;
+    }
+
+    /**
+     * This method is used to fetch all linked space corresponding to a space
+     */
+    @Override
+    public Map<String, List<SpaceLinkDisplay>> getSpaceToSpaceLinks() {
+
+        Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap = new HashMap<>();
+        Iterable<SpaceLinkDisplay> spaceToSpaceLinksList = spaceLinkDisplayRepository.findAll();
+        for (SpaceLinkDisplay link : spaceToSpaceLinksList) {
+            String spaceId = null;
+            if (link.getLink() != null && link.getLink().getSourceSpace() != null) {
+
+                spaceId = link.getLink().getSourceSpace().getId();
+                if (!spaceToSpaceLinksMap.containsKey(spaceId)) {
+                    spaceToSpaceLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToSpaceLinksMap.get(spaceId).add(link);
+            }
+        }
+        return spaceToSpaceLinksMap;
+    }
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space
+     * 
+     * @param spaceLinkMap          For holding list of all links corresponding to
+     *                              each space
+     * @param spaceToModuleLinksMap For getting list of links from one space to
+     *                              other modules
+     * @param spaceToSpaceLinksMap  For getting list of links from one space to
+     *                              other spaces
+     */
+    @Override
+    public Map<String, List<String>> getSpaceLinkMap(Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap,
```

Comment:
what's the difference between this method and `getSpaceToSpaceLinks`? This needs to be clear from method name and javadoc.